torch/_inductor/memory.py

```diff
@@ -0,0 +1,543 @@
+from __future__ import annotations
+
+import dataclasses
+import heapq
+import logging
+from typing import Callable, Dict, List, Set, Tuple, TYPE_CHECKING, Union
+
+from torch.utils._ordered_set import OrderedSet
+
+from .ir import MultiOutputLayout
+from .utils import get_dtype_size
+from .virtualized import V
+
+
+if TYPE_CHECKING:
+    from .dependencies import Dep
+    from .scheduler import BaseSchedulerNode, SchedulerBuffer
+
+
+torch_log = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class MemoryPlanningInfoForBuffer:
+    size_alloc: int = 0
+    size_free: int = 0
+    outdegree: int = 0
+    succ_nodes: OrderedSet[BaseSchedulerNode] = dataclasses.field(
+        default_factory=OrderedSet
+    )
+
+
+@dataclasses.dataclass
+class MemoryPlanningInfoForNode:
+    pred_buffers: List[Union[SchedulerBuffer, FreeableInputBuffer]] = dataclasses.field(
+        default_factory=list
+    )
+    pred_nodes: List[BaseSchedulerNode] = dataclasses.field(default_factory=list)
+    succ_nodes: List[BaseSchedulerNode] = dataclasses.field(default_factory=list)
+    indegree: int = 0
+    index: int = 0
+    memory_to_free: int = 0
+    size: int = 0
+    measure: int = 0
```

Comment:
is `measure` only used in the DFS algorithm? wonder if we can make this field internal to that algorithm function.

also is there a way to add more details on the meaning of `measure`? (the name might be a bit vague I think)

Reply:
I changed the name from `measure` to `size_with_reads` to make it more clear.